ably/types/stats.py

```diff
@@ -4,137 +4,28 @@
-        kwargs = {k: mc_dict[k] for k in mc_dict if (k in expected)}
-        return MessageCount(**kwargs)
-
-
-class MessageTypes:
-    def __init__(self, all=None, messages=None, presence=None):
-        self.all = all or MessageCount()
-        self.messages = messages or MessageCount()
-        self.presence = presence or MessageCount()
-
-    @staticmethod
-    def from_dict(mt_dict):
-        mt_dict = mt_dict or {}
-        kwargs = {
-            "all": MessageCount.from_dict(mt_dict.get("all")),
-            "messages": MessageCount.from_dict(mt_dict.get("messages")),
-            "presence": MessageCount.from_dict(mt_dict.get("presence")),
-        }
-        return MessageTypes(**kwargs)
-
-
-class MessageTraffic:
-    def __init__(self, all=None, realtime=None, rest=None, webhook=None):
-        self.all = all or MessageTypes()
-        self.realtime = realtime or MessageTypes()
-        self.rest = rest or MessageTypes()
-        self.webhook = webhook or MessageTypes()
-
-    @staticmethod
-    def from_dict(mt_dict):
-        mt_dict = mt_dict or {}
-        kwargs = {
-            "all": MessageTypes.from_dict(mt_dict.get("all")),
-            "realtime": MessageTypes.from_dict(mt_dict.get("realtime")),
-            "rest": MessageTypes.from_dict(mt_dict.get("rest")),
-            "webhook": MessageTypes.from_dict(mt_dict.get("webhook")),
-        }
-        return MessageTraffic(**kwargs)
-
-
-class RequestCount:
-    def __init__(self, succeeded=0, failed=0, refused=0):
-        self.succeeded = succeeded
-        self.failed = failed
-        self.refused = refused
-
-    @staticmethod
-    def from_dict(rc_dict):
-        rc_dict = rc_dict or {}
-        expected = ['succeeded', 'failed', 'refused']
-        kwargs = {k: rc_dict[k] for k in rc_dict if (k in expected)}
-        return RequestCount(**kwargs)
-
-
 class Stats:
 
-    def __init__(self, all=None, inbound=None, outbound=None, persisted=None,
-                 connections=None, channels=None, api_requests=None,
-                 token_requests=None, interval_granularity=None,
-                 interval_id=None):
-        self.all = all or MessageTypes()
-        self.inbound = inbound or MessageTraffic()
-        self.outbound = outbound or MessageTraffic()
-        self.persisted = persisted or MessageTypes()
-        self.connections = connections or ConnectionTypes()
-        self.channels = channels or ResourceCount()
-        self.api_requests = api_requests or RequestCount()
-        self.token_requests = token_requests or RequestCount()
+    def __init__(self, entries=None, unit=None, interval_id=None, in_progress=None, app_id=None, schema=None):
         self.interval_id = interval_id or ''
-        self.interval_granularity = (interval_granularity or
-                                     granularity_from_interval_id(self.interval_id))
+        self.entries = entries
+        self.unit = (unit or granularity_from_interval_id(self.interval_id))
```

Comment:
Is this intentional? `TS15b` says that `unit` is meant to come straight from the JSON.

Reply:
ah my bad, this wasn't intentional - was just left over from the previous `granularity` field, i've updated it now to only use the json